### PR TITLE
Add support of Azure asset IDs for the vector store.

### PR DESCRIPTION
**Pull request by nick863**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

fix https://github.com/Azure/azure-rest-api-specs/issues/31349

### CODE REVIEW

**Comment by jhakulin** [comment]
Reviewing soon, thank you

File: specification/ai/Azure.AI.Projects/agents/messages/models.tsp [hunk 1]
```diff
@@ -92,7 +92,12 @@ model ThreadMessage {
 model MessageAttachment {
   /** The ID of the file to attach to the message. */
   @encodedName("application/json", "file_id")
-  fileId: string;
+  fileId?: string;
+
+  /** The data sorces to be used. This option is mutually exclusive with fileId. */
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/messages/models.tsp:97]: sorces -> sources

Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/messages/models.tsp]: Fixed.

File: specification/ai/Azure.AI.Projects/agents/tools/tool_resources.tsp [hunk 2]
```diff
@@ -54,6 +54,11 @@ model CodeInterpreterToolResource {
   @encodedName("application/json", "file_ids")
   @maxItems(20)
   fileIds?: string[] = #[];
+
+  /** The data sources to be used. This option is mutually exclusive with fileIds. */
+  @maxItems(20)
+  @encodedName("application/json", "data_sources")
+  dataSources?: VectorStorageDataSource[];
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/tools/tool_resources.tsp:61]: does the docs gets generated here?

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 3]
```diff
@@ -87,3 +87,44 @@ model VectorStoreStaticChunkingStrategyResponse
...
+/** Type of vector storage configuration. */
+union VectorStorageDataSourceAssetType {
+  uri_asset: "uri_asset",
+  id_asset: "id_asset"
+}
+
+/**
+ * The structure to be supplied when the vector store is being created from the Azure asset ID.
+ */
+model VectorStorageDataSource {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:100]: Docs also here

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 4]
```diff
@@ -87,3 +87,44 @@ model VectorStoreStaticChunkingStrategyResponse
...
+model VectorStorageConfiguration {
+  /** Data sources */
+  @encodedName("application/json", "data_sources")
+  dataSources: VectorStorageDataSource[]
+}
+
+/**
+ * Vector store, keeping several vector store configurations.
+ */
+model VectorStoreFromAzure {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:122]: Not sure about VectorStoreFromAzure name, what was the main purpose of this class? We can then think about name,

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 5]
```diff
@@ -87,3 +87,44 @@ model VectorStoreStaticChunkingStrategyResponse
...
+
+  /** The asset type **/
+  @encodedName("application/json", "type")
+  assetType: VectorStorageDataSourceAssetType;
+}
+
+/**
+ * Vector storage configuration represents the collection of data sources.
+ */
+model VectorStorageConfiguration {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:113]: Docs

File: specification/ai/Azure.AI.Projects/agents/tools/tool_resources.tsp (hunk 2 above)
Comment (MUHAMMADIHSANALIKAP) [reply specification/ai/Azure.AI.Projects/agents/tools/tool_resources.tsp:61]: MUHAMMAD IHSAN ALI.

Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/tools/tool_resources.tsp:61]: The comment on line 58 is used as a doc for class and for the variable: https://github.com/Azure/azure-sdk-for-python/blob/39fcce829c41d947b86f5b244865de4fe13ee380/sdk/ai/azure-ai-projects/azure/ai/projects/models/_models.py#L645

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp (hunk 5 above)
Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:115]: Modified the docstring.

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 6]
```diff
@@ -87,3 +87,46 @@ model VectorStoreStaticChunkingStrategyResponse
...
+  /** Data sources */
+  @encodedName("application/json", "data_sources")
+  dataSources: VectorStorageDataSource[]
+}
+
+/**
+ * The structure, containing the list of vector storage configurations i.e. the list of azure asset IDs.
+ */
+model VectorStoreAzureConfigurations {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:124]: Please rename to VectorStoreConfigurations

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 7]
```diff
@@ -87,3 +87,46 @@ model VectorStoreStaticChunkingStrategyResponse
...
+  /** The asset type **/
+  @encodedName("application/json", "type")
+  assetType: VectorStorageDataSourceAssetType;
+}
+
+/**
+ * Vector storage configuration is the list of data sources, used when multiple
+ * files can be used for the enterprise file search.
+ */
+model VectorStorageConfiguration {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:115]: please rename to VectorStoreConfiguration

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 8]
```diff
@@ -87,3 +87,46 @@ model VectorStoreStaticChunkingStrategyResponse
...
+union VectorStorageDataSourceAssetType {
+  uri_asset: "uri_asset",
+  id_asset: "id_asset"
+}
+
+/**
+ * The structure, containing Azure asset ID and the asset type of the file used as a data source
+ * the enterprise file search.
+ */
+model VectorStorageDataSource {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:101]: please rename to VectorStoreDataSource to be consistent with other VectorStore*

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 9]
```diff
@@ -87,3 +87,46 @@ model VectorStoreStaticChunkingStrategyResponse
...
+  /** Asset URI. */
+  @encodedName("application/json", "uri")
+  storageUri: string;
+
+  /** The asset type **/
+  @encodedName("application/json", "type")
+  assetType: VectorStorageDataSourceAssetType;
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:108]: please rename to VectorStoreDataSourceAssetType

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp [hunk 10]
```diff
@@ -87,3 +87,46 @@ model VectorStoreStaticChunkingStrategyResponse
   /** The options for the static chunking strategy. */
   static: VectorStoreStaticChunkingStrategyOptions;
 }
+
+/** Type of vector storage configuration. */
+union VectorStorageDataSourceAssetType {
```
Comment (jhakulin) [inline specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp:92]: please rename to VectorStoreDataSourceAssetType

**Review by jhakulin — APPROVED** [review]

File: specification/ai/Azure.AI.Projects/agents/tools/models.tsp [hunk 11]
```diff
@@ -95,6 +95,22 @@ model FileSearchToolDefinitionDetails {
...
   maxNumResults?: int32;
+
+  @encodedName("application/json", "ranking_options")
+  rankingOptions?: RankingOptions
+}
+
+/**
+ * Ranking options for file search.
+ */
+model RankingOptions {
```
[the remark this answers is not in the document]
Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/tools/models.tsp]: Fixed.

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp (hunk 10 above)
Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp]: Fixed

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp (hunk 8 above)
Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp]: Fixed.

File: specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp (hunk 6 above)
Comment (nick863) [reply specification/ai/Azure.AI.Projects/agents/vector_stores/common/main.tsp]: Fixed.

**Review by dargilco — APPROVED** [review]

**Pull request merged**